deployment-pipeline.md

```diff
@@ -0,0 +1,90 @@
+
+  If you use modern RN, you probably don't need to run `react-native link` command and you don't have react-native installed globally.
+
+  That's why at the end of `yo rn-toolbox:fastlane-env` command you will get this error:
+
+  ```
+  Linking AppCenter CodePush for you...
+  PLEASE LEAVE THE KEY FIELDS BLANK.
+  Configuration went well but there was an error while installing AppCenter CodePush. Please install it manually with `react-native link react-native-code-push`. Then rerun the generator.
+  Error: spawnSync react-native ENOENT
+  ```
+
+  This is because "fastlane-env" script has `react-native link` commands, and we can simply remove them.
+
+  To do so, navigate to `/Users/{userName}/.config/yarn/global/node_modules/generator-rn-toolbox/generators/fastlane-env/index.js` and replace `_reactNativeLinkDependencies()` with this function:
+
+  ``` js
+  _reactNativeLinkDependencies() {
+      if (this.answers.useAppcenterSDK) {
+        this.log('Installing CocoaPods (please provide sudo password)...');
+        this._runInstallCommandsWithErrorMessages(
+          cocoaPodsInstallCommands,
+          'CocoaPods',
+          'sudo gem install cocoapods && cd ios && pod init && pod repo update.'
+        );
+      }
+  }
+  ```
+
+* Fix for `bundle exec fastlane ios setup --env=staging`.
+  
+  Just remove '=' sign. Should be `bundle exec fastlane ios setup --env staging`
+
+* 2FA in your Apple ID while running in CI.
+
+  If you have 2FA in your account, command will ask you to write the code from SMS. It's not a problem if you run script locally, but in CI you can't write it.
+
+  You can fix it by adding apple_id (id of the app from AppStore) to the pilot():
+
+  ```
+  lane :deploy_to_testflight do |options|
+    pilot(
+      username: ENV['IOS_APPSTORECONNECT_USER_ID'],
+      app_identifier: ENV['IOS_APP_ID'],
+      apple_id: '1540297681',                  <---- here
+      ipa: ENV['IOS_IPA_PATH'],
+      distribute_external: false,
+      skip_waiting_for_build_processing: true
+    )
+  end
+  ```
+
+* AppCenter suggestion
+  
+  You can omit the creation of the production variant of the app in AppCenter if you will use TestFlight and internal testing.
+
+
+# Example app (Sandbox Mobile App)
+
+While learning this, I created the sandbox app for test purposes. You can use it as a reference to check configuration files, etc.
+
+[Github](https://github.com/uptechteam/Sandbox-Mobile-App)
+
+[Github repo for Match](https://github.com/uptechteam/Sandbox-Mobile-App-ios-certs)
+
+You can also find it in `Google Play Console`, `Firebase`, `AppCenter`.
+
+Because of some problem with Notification setups (related to Apple Developer Keys) - I created the `iOS` version under my personal App Store Connect account. So, if you will need access to it - just send me PM (Dmitry Rudiuk).
+
+>* Match Repo pass: `Dimon4350489` (yeah, I know, the dumb password, but I can't change it now in an easy way, sorry)
```

Comment:
hahah, I think no need for `(yeah, I know, the dumb password, but I can't change it now in an easy way, sorry)` part, password is a password, nothing dumb with it :)

Reply:
=D
Thanks for your review!